app/util/aggregation-batch.ts

```diff
@@ -182,28 +181,40 @@ async function createStacCatalogForBatch(
  * @param tx - the database transaction
  * @param workflowStep- the step in the workflow that needs batching
  * @param batch - the Batch to process
+ * @param logger - the Logger for the request
+ *
+ * @returns true if a new work item was created
  */
-async function createCatalogAndWorkItemForBatch(tx: Transaction, workflowStep: WorkflowStep, batch: Batch): Promise<void> {
+async function createCatalogAndWorkItemForBatch(
+  tx: Transaction, workflowStep: WorkflowStep, batch: Batch, logger: Logger,
+): Promise<boolean> {
   const { jobID, serviceID, stepIndex } = workflowStep;
   const batchItemUrls = await getItemUrlsForJobServiceBatch(tx, jobID, serviceID, batch.batchID);
-  // create STAC catalog for the batch
-  const catalogUrl = await createStacCatalogForBatch(
-    jobID,
-    stepIndex,
-    batch.batchID,
-    workflowStep.collectionsForOperation()[0],
-    batchItemUrls);
+  if (batchItemUrls && batchItemUrls.length > 0) {
```

Comment:
Not sure if it is actually better, but I think you can use
```
if (batchItemUrls?.length > 0) {
```
here as well

Reply:
I started with that that but didn't like it because I'm doing the comparison with 0, but yes the code did work.